internal/provider/kubernetes/controller.go

```diff
@@ -495,6 +495,14 @@ func (r *gatewayAPIReconciler) statusUpdateForGateway(gtw *gwapiv1b1.Gateway, sv
 			}
 			gCopy := g.DeepCopy()
 			gCopy.Status.Conditions = status.MergeConditions(gCopy.Status.Conditions, gtw.Status.Conditions...)
+			for index := range gCopy.Status.Conditions {
+				gCopy.Status.Conditions[index].ObservedGeneration = gtw.Generation
```

Comment:
this loop needs to be removed, this should happen automatically

Reply:
After removing them, CI fails : (